staging/src/k8s.io/dynamic-resource-allocation/resourceslice/resourceslicecontroller.go

```diff
@@ -284,24 +369,26 @@ func newController(ctx context.Context, options Options) (*Controller, error) {
 		driverName:       options.DriverName,
 		owner:            options.Owner.DeepCopy(),
 		queue:            options.Queue,
-		resources:        options.Resources.DeepCopy(),
 		mutationCacheTTL: ptr.Deref(options.MutationCacheTTL, defaultMutationCacheTTL),
 		syncDelay:        ptr.Deref(options.SyncDelay, defaultSyncDelay),
+		errorHandler:     options.ErrorHandler,
 	}
 	if c.queue == nil {
 		c.queue = workqueue.NewTypedRateLimitingQueueWithConfig(
 			workqueue.DefaultTypedControllerRateLimiter[string](),
 			workqueue.TypedRateLimitingQueueConfig[string]{Name: "node_resource_slices"},
 		)
 	}
+	if c.errorHandler == nil {
+		c.errorHandler = func(ctx context.Context, err error, msg string) {
+			runtime.HandleErrorWithContext(ctx, err, msg)
+		}
+	}
 	if err := c.initInformer(ctx); err != nil {
 		return nil, err
 	}
 
-	// Sync each desired pool once.
-	for poolName := range options.Resources.Pools {
-		c.queue.Add(poolName)
-	}
+	c.Update(options.Resources)
```

Comment:
Now that we're calling Update() as part of the constructor function we're going to call it twice in a row in `staging/src/k8s.io/dynamic-resource-allocation/kubeletplugin/draplugin.go` via `(d *Helper) PublishResources` when our `d.resourceSliceController` is nil and we need to go through the flow of constructing a new resource slice controller. Is that okay?

Reply:
https://github.com/kubernetes/kubernetes/blob/908bdb3f2c487061975b9a06ca0480e3388bc285/staging/src/k8s.io/dynamic-resource-allocation/kubeletplugin/draplugin.go#L629
ensures that the `PublishResources` is not called when a new controller got constructed, so it won't be done twice.

But this Go pattern of avoiding `else` clauses to reduce indention isn't really doing us any favor here. I've changed it.

Reply:
You're right, I missed the early return! I do think your change is more expressive of the flow, thx.